src/main/native/windows/util.cc

```diff
@@ -92,20 +92,28 @@ bool AutoAttributeList::Create(HANDLE* handles, size_t count,
     return false;
   }
 
+  static constexpr size_t kHandleCount = 3;
+  std::unique_ptr<HANDLE[]> handles(new HANDLE[kHandleCount]);
+  handles[0] = stdin_h;
+  handles[1] = stdout_h;
+  handles[2] = stderr_h;
   if (!UpdateProcThreadAttribute(attrs, 0, PROC_THREAD_ATTRIBUTE_HANDLE_LIST,
-                                 handles, count * sizeof(HANDLE), NULL, NULL)) {
+                                 handles.get(), kHandleCount * sizeof(HANDLE),
+                                 NULL, NULL)) {
     if (error_msg) {
       DWORD err = GetLastError();
       *error_msg = MakeErrorMessage(WSTR(__FILE__), __LINE__,
                                     L"UpdateProcThreadAttribute", L"", err);
     }
     return false;
   }
-  result->reset(new AutoAttributeList(data.release()));
+  result->reset(new AutoAttributeList(&data, &handles));
```

Comment:
Consider using `std::move` so that reader knows this is to transfer ownership.

```suggestion
  result->reset(new AutoAttributeList(std::move(data), std::move(handles)));
```

Then,

```cpp
AutoAttributeList::AutoAttributeList(std::unique_ptr<uint8_t[]>&& data,
                                     std::unique_ptr<HANDLE[]>&& handles)
    : data_(data), handles_(handles) {}
```

I actually would prefer the `AutoAttributeList` to own three `HANDLE` directly instead of storing in `handles_`. Having three `HANDLE` member vars means I don't have to remember what are the `handles_[0/1/2]`. `InitStartupInfoEx{A,W}` will need to manually set the three `HANDLE`s anyway.

Code with less heap alloc is easier to reason with.

Reply:
Thanks for the suggestion!

The HANDLEs must be laid out consecutively in memory, that's how `UpdateProcThreadAttribute` and `CreateProcessW` expects them. If they are just three member variables of the class, then there's a risk of messing up that invariant by adding more member variables between them.

How about introducing a struct containing three HANDLEs? That's less likely to get messed up like I described.

Reply:
> The HANDLEs must be laid out consecutively in memory, that's how UpdateProcThreadAttribute and CreateProcessW expects them. If they are just three member variables of the class, then there's a risk of messing up that invariant by adding more member variables between them.

In that case, maybe just store them as `HANDLE handles_[3]` (so you can't have other member variables in between) and `memcpy` in constructor?

Reply:
See new commit.

Reply:
New commit looks good. Ignore https://github.com/bazelbuild/bazel/pull/6758#discussion_r236266198